powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/model/EciesCryptogram.java

```diff
@@ -28,6 +28,28 @@ public class EciesCryptogram {
     private final byte[] mac;
     private final byte[] encryptedData;
     private final byte[] nonce;
+    private final byte[] associatedData;
+    private final byte[] timestamp;
```

Comment:
I think that it would be better to use `Long` here and do the conversion to bytes internally, if this is the object we fill directly from the request object or use for the response. We can also use Builder, as @banterCZ  suggested, with `Long` at input and keep the bytes in the object.

Reply:
Ok, I will switch to `Long`.